crates/api/src/proposer/api.rs

```diff
@@ -1313,6 +1274,29 @@ where
 
         // Update duties if applicable
         if let Some(new_duties) = slot_update.new_duties {
+            for duty in new_duties.iter() {
+                match self.delegation_contract.get_preconfer_election(&duty.entry.registration.message.public_key, duty.slot).await {
+                    Ok(Some(elected_preconfer)) => match self.auctioneer.save_new_gateway_election(&elected_preconfer, duty.slot).await {
```

Comment:
We need to check if the elected_preconfer is changed before saving to db.

Reply:
address this in another pr https://github.com/lu-bann/helix/issues/25